userdata/rundsc.ps1

```diff
@@ -908,12 +912,25 @@ if ($rebootReasons.length) {
         }
       }
       if ((@(Get-Process | ? { $_.ProcessName -eq 'generic-worker' }).length -eq 0)) {
-        Write-Log -message 'no generic-worker process detected.' -severity 'INFO'
-        & format @('Z:', '/fs:ntfs', '/v:""', '/q', '/y')
-        Write-Log -message 'Z: drive formatted.' -severity 'INFO'
-        #& net @('user', 'GenericWorker', (Get-ItemProperty -path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon' -name 'DefaultPassword').DefaultPassword)
-        Remove-Item -Path $lock -force -ErrorAction SilentlyContinue
-        & shutdown @('-r', '-t', '0', '-c', 'reboot to rouse the generic worker', '-f', '-d', 'p:4:1') | Out-File -filePath $logFile -append
+        if ($locationType -eq 'AWS') { 
+          Write-Log -message 'no generic-worker process detected.' -severity 'INFO'
+          & format @('Z:', '/fs:ntfs', '/v:""', '/q', '/y')
+          Write-Log -message 'Z: drive formatted.' -severity 'INFO'
+          #& net @('user', 'GenericWorker', (Get-ItemProperty -path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon' -name 'DefaultPassword').DefaultPassword)
+          Remove-Item -Path $lock -force -ErrorAction SilentlyContinue
+          & shutdown @('-r', '-t', '0', '-c', 'reboot to rouse the generic worker', '-f', '-d', 'p:4:1') | Out-File -filePath $logFile -append
+        }
+        if ($locationType -eq 'DataCenter') {
+          if (!(Test-Path "$DC1stRunLock")) {
+            Remove-Item -Path $lock -force -ErrorAction SilentlyContinue
+            & shutdown @('-r', '-t', '0', '-c', 'reboot to rouse the generic worker', '-f', '-d', 'p:4:1') | Out-File -filePath $logFile -append
+          }
+          # For the rare case of a datacenter machine making it this far without Generic User logged in
+          $CurrentUser = gwmi -Class win32_computersystem -ComputerName localhost | select -ExpandProperty username -ErrorAction Stop 
+          if ($CurrentUser -notcontains "Generic") {
+            shutdown @('-r', '-t', '0', '-c', 'Generic Worker failed to log in', '-f', '-d', 'p:4:1') | Out-File -filePath $logFile -append
```

Comment:
For generic worker 10 onwards, there is no GenericWorker user, instead the worker runs as LocalSystem in a windows service, so I think this needs some adjustment?